std/meta.d

```diff
@@ -863,6 +863,97 @@ unittest
     }
 }
 
+/**
+ * Converts an input range $(D range) to an alias sequence.
+ */
+template aliasSeqOf(alias range)
+{
+    import std.range : isInputRange;
+    import std.traits : isArray, isNarrowString;
+
+    alias ArrT = typeof(range);
+    static if (isArray!ArrT && !isNarrowString!ArrT)
+    {
+        static if (range.length == 0)
+        {
+            alias aliasSeqOf = AliasSeq!();
+        }
+        else static if (range.length == 1)
+        {
+            alias aliasSeqOf = AliasSeq!(range[0]);
+        }
+        else
+        {
+            alias aliasSeqOf = AliasSeq!(aliasSeqOf!(range[0 .. $/2]), aliasSeqOf!(range[$/2 .. $]));
+        }
+    }
+    else static if (isInputRange!ArrT)
+    {
+        import std.array : array;
+        alias aliasSeqOf = aliasSeqOf!(array(range));
+    }
+    else
+    {
+        import std.string : format;
+        static assert(false, format("Cannot transform %s of type %s into a AliasSeq.", range, ArrT.stringof));
+    }
+}
+
+///
+unittest
+{
+    import std.algorithm : map, sort;
+    import std.string : capitalize;
+
+    struct S
+    {
+        int a;
+        int c;
+        int b;
+    }
+
+    alias capMembers = aliasSeqOf!([__traits(allMembers, S)].sort().map!capitalize());
+    static assert(capMembers[0] == "A");
+    static assert(capMembers[1] == "B");
+    static assert(capMembers[2] == "C");
+}
+
+///
+unittest
+{
+    enum REF = [0, 1, 2, 3];
+    foreach(I, V; aliasSeqOf!([0, 1, 2, 3]))
```

Comment:
Nitpick: `I` and `V` will never refer to types, so they should not be capitalized.